WeakAurasOptions/AuthorOptions.lua

```diff
@@ -0,0 +1,1278 @@
+              local childData = data[childID]
+              local childOption = childData.authorOptions[optionID]
+              local insertPoint = math.min(j, #childOption.values + 1)
+              if value == "" then
+                tremove(childOption.values, insertPoint)
+              else
+                childOption.values[insertPoint] = value
+              end
+              WeakAuras.Add(childData)
+            end
+            WeakAuras.ReloadTriggerOptions(data[0])
+          else
+            if value == "" then
+              tremove(values, j)
+            else
+              values[j] = value
+            end
+            WeakAuras.Add(data)
+            WeakAuras.ReloadTriggerOptions(data)
+          end
+        end
+      }
+      order = order + 1
+      args["option" .. i .. "valdelete" .. j] = {
+        type = "execute",
+        name = "",
+        order = order,
+        func = function()
+          if option.references then
+            for childID, optionID in pairs(option.references) do
+              local childData = data[childID]
+              local childOption = childData.authorOptions[optionID]
+              tremove(childOption.values, j)
+              WeakAuras.Add(childData)
+            end
+            WeakAuras.ReloadTriggerOptions(data[0])
+          else
+            tremove(values, j)
+            WeakAuras.Add(data)
+            WeakAuras.ReloadTriggerOptions(data)
+          end
+        end,
+        width = 0.15,
+        image = "Interface\\AddOns\\WeakAuras\\Media\\Textures\\delete",
+        imageWidth = 24,
+        imageHeight = 24
+      }
+    end
+    args["option" .. i .. "newvaluespace"] = {
+      type = "toggle",
+      name = (values[#values] == conflict and "|cFF4080FF" or "") .. L["Value %i"]:format(#values + 1),
+      order = order,
+      disabled = function() return true end,
+      get = function() return true end,
+      set = function() end,
+    }
+    order = order + 1
+    args["option" .. i .. "newvalue"] = {
+      type = "input",
+      name = "",
+      order = order,
+      get = function() return "" end,
+      set = function(_, value)
+        if option.references then
+          for childID, optionID in pairs(option.references) do
+            local childData = data[childID]
+            local childOption = childData.authorOptions[optionID]
+            childOption.values[#childOption.values + 1] = value
+            WeakAuras.Add(childData)
+          end
+          WeakAuras.ReloadTriggerOptions(data[0])
+        else
+          values[#values + 1] = value
+          WeakAuras.Add(data)
+          WeakAuras.ReloadTriggerOptions(data)
+        end
+      end
+    }
+    order = order + 1
+    return order
+  end
+}
+
+local function up(data, option, index)
+  if option.references then
+    return function()
+      for _, optionID in pairs(option.references) do
+        if index == 1 then return true end
```

Comment:
What about if I have:
A B F
A C F
A D F
A E F

The merged display would look something like:

A B C D E F

And if i moved up C, then this would move up C only in the second item, and produce

C A B D E F

I think that this would be rather surprising if you weren't expecting it, and building something to make it produce A C B D E F would be both fiddly and probably undesirable. Should I just let that slide?

Reply:
Ah, I didn't consider that. Oh well, seems my solution has its own corner cases.